lib/ts/controllers/s-tooltip.ts

```diff
@@ -0,0 +1,141 @@
+
+class TooltipController extends Stacks.StacksController {
```

Comment:
Instead of copy/pasting all the Popover code, can we instead inherit and extend it? Hopefully this isn't too difficult since we're using TS. Will re-review this file after all duped code is removed.